### PR TITLE
atlassian_{confluence,jira}: clarify API authentication

### LINKED ISSUE #3692: [atlassian_jira] Update username / password descriptions for Atlassian Cloud
Configuring the integration for Atlassian Cloud is a bit confusing w.r.t. authentication. When connection to Atlassian Cloud you must put your email address into the username field and your API token into the password field. We should probably rename the fields to be more clear and explicitly state what goes into them.

Screen Shot 2022-07-12 at 20 09 40

https://developer.atlassian.com/cloud/jira/platform/basic-auth-for-rest-apis/

The same changes are needed in `atlassian_confluence`.

### PR DESCRIPTION
**Pull request by efd6**
## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

- Closes #3692

## Screenshots

 
Screen Shot 2022-07-13 at 10 34 58
Screen Shot 2022-07-13 at 10 36 06

### CODE REVIEW

**Review by andrewkroh — COMMENTED** [review]
I was reviewing the README to see if there were any additional changes that could help. I noticed something that makes me question the original intent of the "Personal Access Token". It says

> JIRA Cloud only supports Basic Auth using username and a Personal Access Token.

That makes me think the "Personal Access Token" was intended for use with Atlassian Cloud. If that's the case then perhaps we can just fully remove the "Personal Access Token" and only support Basic Auth for both self-hosted Jira and Atlassian Cloud.

@legoguy1000 Do you recall the intent behind the "Personal Access Token"? Was that for the Atlassian Cloud case?

**Comment by legoguy1000** [comment]
So if I recall everything from my testing, the self hosted confluence and JIRA support basic auth with user/password OR Bearer auth header with a PAT. Cloud versions only supported basic auth with email/token.

**Comment by efd6** [comment]
Is there a decision on this?

**Comment by andrewkroh** [comment]
> the self hosted confluence and JIRA support basic auth with user/password OR Bearer auth header with a PAT

Yeah, it looks to be true that a PAT can be used with self-hosted JIRA via a `Authentication: Bearer <pat>` header. https://developer.atlassian.com/server/jira/platform/rest-apis/

So I think we should clarify the readme by
  - stating the two options for authenticating to "self-hosted" JIRA,
  - stating that only basic auth can be used to Atlassian Cloud (mention the specific field names from the UI),
  - and adding links to Atlassian documentation like
    - https://developer.atlassian.com/server/jira/platform/basic-authentication/
    - https://confluence.atlassian.com/enterprise/using-personal-access-tokens-1026032365.html
    - https://developer.atlassian.com/cloud/jira/platform/basic-auth-for-rest-apis/

In UI description of the "Personal Access Token" I think it should mention that this is only for self-hosted JIRA (or not for Atlassian Cloud). This will help steer users toward the correct fields if the don't read the README.

**Comment by legoguy1000** [comment]
I would make it clear that the PAT in the Auth header is only for self
hosted.  U CAN use a PAT with ur username in Basic auth for cloud.

**Comment by andrewkroh** [comment]
IMO if we removed the abstraction from the UI / manifest.yml then determining what options to set based on use case gets simpler. Like if the UI had three config field, Basic Auth Username, Basic Auth Password, and Bearer Authentication Header then when I'm reading the Atlassian documentation I have a straight forward way to map things into our UI.

The descriptions of the fields can still be there to guide users as to what to enter based on whether that are connecting to Jira Server (self-hosted) or Atlassian Cloud.

File: packages/atlassian_jira/_dev/build/docs/README.md [hunk 1]
```diff
@@ -2,9 +2,15 @@
...
+## Authentication Set-Up
+
+When setting up the Atlassian Jira Integration for Atlassian Cloud you will need to use the "Jira User Identifier" and "Jira API Token" fields in the integration configuration. These will allow connection to the [Atlassian Cloud REST API](https://developer.atlassian.com/cloud/jira/platform/basic-auth-for-rest-apis/) via [Basic Authentication](https://developer.atlassian.com/server/jira/platform/basic-authentication/).
+
+If you are using a self-hosted instance, you will be able to use either the "Jira User Identifier" and "Jira API Token" fields above, *or* use the "Personal Access Token" field to [authenticate with a PAT](https://confluence.atlassian.com/enterprise/using-personal-access-tokens-1026032365.html). If the "Personal Access Token" field is set in the configuration, it will take precedence over the User ID/API Token fields. 
+
 ## Logs
 
-### Audit
+z### Audit
```
Comment (legoguy1000) [inline packages/atlassian_jira/_dev/build/docs/README.md:13]: typo

Comment (efd6) [reply packages/atlassian_jira/_dev/build/docs/README.md:13]: 😆 

The consequence of using hot-row modifiers.

**Review by andrewkroh — APPROVED** [review]

**Pull request merged**